File: JAVA-Basic/JavaBasic/src/WhileLoop/Exercise/Coins.java
package WhileLoop.Exercise;

import java.util.Scanner;

public class Coins {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        double sum = Math.floor(Double.parseDouble(scan.nextLine()) * 100);
        int coins = 0;

        while (sum > 0) {
            if (sum >= 200) {
                sum -= 200;
                coins++;
            } else if (sum >= 100) {
                sum -= 100;
                coins++;
            } else if (sum >= 50) {
                sum -= 50;
                coins++;
            } else if (sum >= 20) {
                sum -= 20;
                coins++;
            } else if (sum >= 10) {
                sum -= 10;
                coins++;
            } else if (sum >= 5) {
                sum -= 5;
                coins++;
            } else if (sum >= 2) {
                sum -= 2;
                coins++;
            } else if (sum >= 1) {
                sum -= 1;
                coins++;
            }
        }
        System.out.println(coins);
    }
}

File: 01.JS-Basic/WHILE-ЦИКЪЛ Упражнение/Coins.js
function coins(input) {

    let sum = Number(input[0])
    let coins = 0

    while (sum != 0) {
        if (sum >= 2) {
            coins++
            sum -= 2
        } else if (sum >= 1) {
            coins++
            sum -= 1
        } else if (sum >= 0.50) {
            coins++
            sum = (sum - 0.50).toFixed(2)
        } else if (sum >= 0.20) {
            coins++
            sum = (sum - 0.20).toFixed(2)
        } else if (sum >= 0.10) {
            coins++
            sum = (sum - 0.10).toFixed(2)
        } else if (sum >= 0.05) {
            coins++
            sum = (sum - 0.05).toFixed(2)
        } else if (sum >= 0.02) {
            coins++
            sum = (sum - 0.02).toFixed(2)
        } else if (sum >= 0.01) {
            coins++
            sum = (sum - 0.01).toFixed(2)
        } else { break }
    }
    console.log(coins)
}
coins(["2.73"])